### ISSUE TITLE
Initialization only partially works, No Commit Tab, but Version Control in Debug Panel

### ISSUE THREAD

**Issue by collectivenectar** [godotengine/godot-git-plugin#87]
I am on OSX Catalina 10.15.7, Macbook Pro 13" 2017 Intel GPU, using Godot v3.3.3 stable [b973f997f]  After each time I do a clean start over I get the same results, so here are the steps I take:

Download the .zip from (https://github.com/godotengine/godot-git-plugin/releases)
place godot-git-plugin folder inside projectfolder/addons/
run godot project
project -> version control -> set up version control
Version Control System -> GitAPI
Initialize

As soon as I click initialize, the 'Version Control' tab shows up in the debug panel to the right of 'Animation', but no commit panel appears to the right of 'Import.' Version Control tab has the refresh button next to diffs, but it's not functional. I tested adding a random var and saving, refreshed, and no diffs detected.

Now, when I did this process the first time, I did have an issue with gatekeeper, and while this thread provided some steps, I did not see the same popup window and did not have the same options:

https://github.com/godotengine/godot-git-plugin/issues/58#issuecomment-793451881

Instead of that popup window, as soon as I ran Godot, OSX told me that the dylib file was not able to be scanned for malware. I then opened system preferences -> Security & Privacy, and there underneath the options to choose from 'Allow apps downloaded from' was a message asking if I wanted to allow the dylib file to be opened. I then hit allow, closed the project and godot, and tried to initialize the version control again, with the same results. 

I tried codesign --force -deep -s -(the dylib file path), as well as code signing all the other files in the plugin folder, I tried changing quarantine attributes (xattr) for the dylib, and nothing had an affect. I spoke with @ ChronicallySerious about the issue and they suggested I try running Godot through terminal to see if there are any error messages that appear. I had a lot of trouble getting godot.app to open from terminal, but finally succeeded after using (echo 'alias godot="/Applications/Godot.app/Contents/MacOS/Godot"' >> ~/.zprofile && source .zprofile && which godot) but there turned out to be no errors when initializing.

I tried a different blank test project, and I tried signing in on a guest profile and copying the procedure, and still I get the same results.  Now that I did all that funky code signing and zbash stuff, if I try a clean start with a new project, gatekeeper doesn't even pop up any more regarding the dylib file, but still same results, so I don't know how I'd be able to get back to reproducing the exact popups or anything.

I'm just going to use git via terminal in the mean time, which I needed to learn anyways, so it's not really an inconvenience to me.

If there are any other things I can try let me know, I've only been programming for about a year so there may be many things that will go over my head, but I'm willing to learn if you can be patient with me. After I shared results w/ @ ChronicallySerious, they suggested I open a bug issue, so here I am.

**Comment by bruvzg** [comment]
> Instead of that popup window, as soon as I ran Godot, OSX told me that the dylib file was not able to be scanned for malware. I then opened system preferences -> Security & Privacy, and there underneath the options to choose from 'Allow apps downloaded from' was a message asking if I wanted to allow the dylib file to be opened. I then hit allow, closed the project and godot, and tried to initialize the version control again, with the same results.

That's expected from unsigned plugin, after clicking "Allow anyway" in the system properties, on the second start a new warning popup with the "Open" button should be displayed, after clicking "Open" it should work.

I have tested v1.2.2 plugin on M1 MacBook running Big Sur 11.6, and it seems to work fine with the Godot 3.3.3 and 3.3.4 stable (tested VCS init and committing changes on fresh empty project).

If you do not get `dylib` loading errors in the terminal, it's probably loaded successfully and something else is wrong.

Try running Godot with the `--verbose` flag to get extra info:
```sh
/Applications/Godot.app/Contents/MacOS/Godot --path YOUR_PROJECT_PATH --editor --verbose
```

**Comment by collectivenectar** [comment]
ran 

```/Applications/Godot.app/Contents/MacOS/Godot --path /my/path/to/godot/project --editor --verbose```

first time but realized I hadn't moved the addons folder in yet. I dragged the addons folder into the project folder, and then closed godot. saw this bit in the console 

```WARNING: cleanup: ObjectDB instances leaked at exit (run with --verbose for details).
   At: core/object.cpp:2132.
Leaked instance: GDNative:21014
Leaked instance: ConfigFile:21006
Leaked instance: GDNativeLibrary:21005 - Resource path: res://addons/godot-git-plugin/git_api.gdnlib
Hint: Leaked instances typically happen when nodes are removed from the scene tree (with `remove_child()`) but not freed (with `free()` or `queue_free()`).
ERROR: clear: Resources still in use at exit (run with --verbose for details).
   At: core/resource.cpp:450.
Resource still in use: res://addons/godot-git-plugin/git_api.gdnlib (GDNativeLibrary)
Orphan StringName: GDNativeLibrary
Orphan StringName: GDNative
Orphan StringName: ConfigFile
StringName: 3 unclaimed string names at exit.
libc++abi.dylib: terminating with uncaught exception of type std::__1::system_error: recursive_mutex lock failed: Invalid argument
zsh: abort      /Applications/Godot.app/Contents/MacOS/Godot --path  --editor --verbose ``` 



and then ran the same verbose command again. got this



 ```Godot Engine v3.3.3.stable.official.b973f997f - https://godotengine.org
Using GLES3 video driver
OpenGL debugging not supported!
OpenGL ES 3.0 Renderer: Intel(R) Iris(TM) Plus Graphics 640
OpenGL ES Batching: ON
	OPTIONS
	max_join_item_commands 16
	colored_vertex_format_threshold 0.25
	batch_buffer_size 16384
	light_scissor_area_threshold 1
	item_reordering_lookahead 4
	light_max_join_items 32
	single_rect_fallback False
	debug_flash False
	diagnose_frame False
CoreAudio: detected 2 channels
CoreAudio: audio buffer frames: 512 calculated latency: 11ms
 
Loading resource: res://addons/godot-git-plugin/git_api.gdnlib
Registered camera FaceTime HD Camera with id 1 position 0 at index 0
CORE API HASH: 15198786384277183696
EDITOR API HASH: 1752727588889334904
Loading resource: res://default_env.tres
Construct gdnative interface

Destruct gdnative interface

Class 'BulletPhysicsDirectBodyState' is not exposed, skipping.
Class 'BulletPhysicsDirectSpaceState' is not exposed, skipping.
Class 'BulletPhysicsServer' is not exposed, skipping.
Class 'GDScriptNativeClass' is not exposed, skipping.
Class 'IP_Unix' is not exposed, skipping.
Class 'InputDefault' is not exposed, skipping.
Class 'Physics2DDirectBodyStateSW' is not exposed, skipping.
Class 'Physics2DDirectSpaceStateSW' is not exposed, skipping.
Class 'Physics2DServerSW' is not exposed, skipping.
Class 'ResourceImporterMP3' is not exposed, skipping.
Class 'ResourceImporterOGGVorbis' is not exposed, skipping.
Loading resource: /Users/jonathan/Library/Application Support/Godot/editor_settings-3.tres
EditorSettings: Load OK!
Loading resource: res://scripts/global.gd
Loaded builtin certs
EditorSettings: Save OK!
Loading resource: res://addons/godot-git-plugin/git_api.gdns
Loading resource: res://scripts/mainmenu.gd
Loading resource: res://scenes/DeckIntentions/deckcorepopup.tscn
Loading resource: res://styleboxes/modalpopupsub.tres
Loading resource: res://styleboxes/modalpopup.tres
Loading resource: res://scripts/DeckIntentions/deckcorepopup.gd
Loading resource: res://app icons/Camera.png
Loading resource: res://shaders/circlemask.tres
Loading resource: res://app icons/BackArrow.png
Loading resource: res://app icons/Swap.png
Loading resource: res://app icons/Cancel.png
Loading resource: res://scenes/UserHistory/UserHistory.tscn
Loading resource: res://fonts/AbrilFatface-Regular.otf
Loading resource: res://scripts/UserHistory/UserHistory.gd
Loading resource: res://scenes/UserHistory/loadareadingpanel.tscn
Loading resource: res://scripts/UserHistory/loadareadingpanel.gd
Loading resource: res://scenes/Guidebook/Guidebook.tscn
Loading resource: res://scenes/DrawingCards/choosewhichspread.tscn
Loading resource: res://scripts/DrawingCards/choosewhichspread.gd
Loading resource: res://scenes/DrawingCards/loadaspreadcontainer.tscn
Loading resource: res://scripts/DrawingCards/choosewhichspreadcontainer.gd
Loading resource: res://scenes/DeckCarousel/DeckCarousel.tscn
Loading resource: res://scripts/DrawingCards/Carousels/DeckCarousel.gd
Loading resource: res://scenes/DrawingCards/Carousels/JournalCarousel.tscn
Loading resource: res://scripts/DeckCarousel/JournalCarousel.gd
Loading resource: res://scenes/Card.tscn
Loading resource: res://scenes/assets/card/cardfacewithoutnumberandrh.png
Loading resource: res://fonts/Kollektif.ttf
Loading resource: res://scenes/assets/card/rainbowharmony.png
Loading resource: res://scenes/assets/card/therealreal.png
Loading resource: res://scenes/assets/card/dash.png
Loading resource: res://scenes/assets/card/flower.png
Loading resource: res://fonts/Kollektif-Bold.ttf
Loading resource: res://scenes/Card.gd
Loading resource: res://shaders/card.shader
Loading resource: res://scenes/cardpanelstyleboxfrontandback.tres
Loading resource: res://scenes/DeckCarousel/cardnotecell.tscn
Loading resource: res://scripts/DeckCarousel/cardnotecell.gd
Loading resource: res://app icons/Add.png
Loading resource: res://app icons/Edit.png
Loading resource: res://styleboxes/cardnotecell.tres
Loading resource: res://app icons/ChevronUp.png
Loading resource: res://app icons/ChevronDown.png
Loading resource: res://scenes/mainmenu.tscn
Loading resource: res://app icons/Save.png
Loading resource: res://app icons/Shuffle.png
Loading resource: res://scripts/mainmenubar.gd
Loading resource: res://app icons/Cards.png
Loading resource: res://app icons/Journal.png
Loading resource: res://addons/godot-git-plugin/git_api.gdns
Loading resource: res://addons/godot-git-plugin/git_api.gdns
Loading resource: res://addons/godot-git-plugin/git_api.gdns
EditorSettings: Save OK!```

That last was what I got after opening godot after the addons folder has been placed into project folder, and then attempting to set up verson control. After selecting initialize, I saw the Version Control tab pop into place, but still no commit tab. I closed the window, and exited Godot, and that's it. Hope my info helps

**Comment by ChronicallySerious** [comment]
The Godot 3.5 beta1 is now live. And along with it, we released a v2.0.0 of this plugin which should fix the leaked instances. Could you please check if you still face this issue and reopen if the problem still lingers?
